DragonSlayer.cpp

```diff
@@ -1,21 +1,35 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer(std::string name_, int hp_, int armor_) : Character(hp_, armor_, 4), name(name_) 
+{
+    helpfulItems = makeHelpfulItems(3);
+    defensiveItems = makeDefensiveItems(1);
+}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return  name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
+        
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //note: they should only use the item if the dragon's hitpoints are > 0...
         //note: items are single-use only, so you need to reset it after use.  
         //look in the Character class for how the other item types are reset after use.
+        if (dragon->getHP() > 0)
+        {
+            attackItem->use(this);
+            attackItem.reset();
+        }
```

Comment:
You need to make sure that the attack item is not null before you use it.
- [ ] revise